tools/bin/mtest

```diff
@@ -498,73 +530,96 @@ def configure_options(target):
-            if path.split("/")[-1].startswith("crypto_config_"):
-                dest = CRYPTO_CONFIG_FILE
-            else:
-                dest = CONFIG_FILE
-            log(f"cp {path[len(ROOT) + 1:] if path.startswith(ROOT) else path} {dest[len(ROOT) + 1:]}")
-            shutil.copy(path, dest)
+
+    for config in header_options:
+        # a config file - overwrite existing config file
+        path = config if os.path.exists(config) else f"configs/{config}"
+        if path.split("/")[-1].startswith("crypto_config_"):
+            new_crypto_config = path
         else:
-            # a configuration recognised by config.py, e.g. "baremetal"
-            log(f"scripts/config.py {config}")
-            subrun([f"{ROOT}/scripts/config.py", config], silent=True)
-
-    if disable_spm:
-        # at some point in the past, MBEDTLS_HAVE_ASM also needs disabling,
-        # and more options again (not sure which) for 3.0.0
-        set_config_option("unset", "MBEDTLS_PSA_CRYPTO_SPM", silent=False)
-    if set_no_platform_entropy:
-        # this gets disabled by #8164 - ensure it is set
-        set_config_option("set", "MBEDTLS_NO_PLATFORM_ENTROPY", silent=False)
+            new_config = path
+        if any(x[0] == os.path.basename(path) for x in ["config-tfm.h", "tfm_mbedcrypto_config_profile_medium", "crypto_config_profile_medium.h"]):
+            config_is_tfm = True
+
+    # backup and overwrite the config files
+    if new_config is not None:
+        backup_config_options(CONFIG_FILE)
+        log(f"cp {new_config} {CONFIG_FILE}")
+        shutil.copy(new_config, CONFIG_FILE)
+    if new_crypto_config is not None:
+        crypto_config_file = get_crypto_config_file()
+        backup_config_options(crypto_config_file)
+        log(f"cp {new_crypto_config} {crypto_config_file}")
+        shutil.copy(new_crypto_config, crypto_config_file)
+    
+    sync()
+
+    # apply bulk config.py settings to main config file
+    for config in config_py_groups:
+        options.remove(config)
+        # a configuration recognised by config.py, e.g. "baremetal"
+        backup_config_options(CONFIG_FILE)
+        log(f"scripts/config.py {config}")
+        subrun([f"scripts/config.py", config], silent=True)
+        
+    if config_is_tfm:
+        # adjust options for older versions of the config files to allow TF-M config to build
+        # at some point in the past, MBEDTLS_HAVE_ASM also needs disabling
+        if not git_history_contains("51e72456f9dd7bfc3a4eee53de046f4390a3d83b"):
+            if target.baremetal:
+                set_config_option("set", "MBEDTLS_NO_PLATFORM_ENTROPY", crypto_config_file, silent=False)
+            if not git_history_contains("0c98f9f8423c74fcd2a04a0f1c17e3f9fb37448b"):
+                set_config_option("unset", "MBEDTLS_PSA_CRYPTO_SPM", crypto_config_file, silent=False)
+                if not git_history_contains("b873577fc341098bab9b96e38c24bc41e0bc784e"):
```

Comment:
This commit is a merge commit - can you use the more relevant commit to make it simpler for reviewers to check? :)